resources/js/oauth2/profile/edit_client/components/oauth_panel.js

```diff
@@ -0,0 +1,377 @@
+                        errors={formik.errors.website}
+                        onChange={formik.handleChange}
+                    />
+                    <SimpleTextFormControl
+                        id="logo_uri"
+                        title="Application Logo Url (optional)"
+                        tooltip="URL that references a logo for the Client application."
+                        type="url"
+                        value={formik.values.logo_uri}
+                        touched={formik.touched.logo_uri}
+                        errors={formik.errors.logo_uri}
+                        onChange={formik.handleChange}
+                    />
+                    <SimpleTextFormControl
+                        id="tos_uri"
+                        title="Application Term of Service Url (optional)"
+                        tooltip="URL that the Relying Party Client provides to the End-User to read about the Relying Party's terms of service."
+                        type="url"
+                        value={formik.values.tos_uri}
+                        touched={formik.touched.tos_uri}
+                        errors={formik.errors.tos_uri}
+                        onChange={formik.handleChange}
+                    />
+                    <SimpleTextFormControl
+                        id="policy_uri"
+                        title="Application Policy Url (optional)"
+                        tooltip="URL that the Relying Party Client provides to the End-User to read about the how the profile data will be used."
+                        type="url"
+                        value={formik.values.policy_uri}
+                        touched={formik.touched.policy_uri}
+                        errors={formik.errors.policy_uri}
+                        onChange={formik.handleChange}
+                    />
+                    <FormControl variant="outlined" className={styles.form_control}>
+                        <FormLabel htmlFor="contacts">
+                            <Typography variant="subtitle2" display="inline">Contact Emails
+                                (optional)</Typography>&nbsp;
+                            <Tooltip title="e-mail addresses of people responsible for this Client.">
+                                <InfoOutlinedIcon fontSize="small"/>
+                            </Tooltip>
+                        </FormLabel>
+                        <TagsInput
+                            id="contacts"
+                            name="contacts"
+                            fullWidth
+                            size="small"
+                            variant="outlined"
+                            type="email"
+                            onChange={formik.handleChange}
+                            tags={getTags(formik.values.contacts)}
+                        />
+                    </FormControl>
+                    {
+                        appType === appTypes.Service &&
```

Comment:
please review the original logic
https://github.com/OpenStackweb/openstackid/blob/5e61b0054c83226b802e4f2f3072087717d8a530/resources/views/oauth2/profile/edit-client-data.blade.php#L155